# Processing X-matched Data

## Loading Data

In [28]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
pd.set_option('display.max_columns', 100)
from astropy import units as u
from tqdm import tqdm

In [42]:
main_df = pd.read_csv('cut_desxwisexvhs.csv', index_col=0)

In [43]:
main_df.columns

Index(['t1_t1_ra', 't1_t1_dec', 't1_t1_mag_aper_4_g', 't1_t1_mag_aper_4_r',
       't1_t1_mag_aper_4_i', 't1_t1_mag_aper_4_z', 't1_t1_mag_aper_4_y',
       't1_t1_magerr_aper_4_g', 't1_t1_magerr_aper_4_r',
       't1_t1_magerr_aper_4_i', 't1_t1_magerr_aper_4_z',
       't1_t1_magerr_aper_4_y', 't1_dec', 't1_dfluxlbs_w1', 't1_dfluxlbs_w2',
       't1_fluxlbs_w1', 't1_fluxlbs_w2', 't1_mag_w1_vg', 't1_mag_w2_vg',
       't1_ra', 't1_dist_arcsec', 'dec2000', 'japermag4', 'japermag4err',
       'ksapermag4', 'ksapermag4err', 'ra2000', 'dist_arcsec'],
      dtype='object')

In [44]:
# Renaming columns for convenience
neat_df = main_df.rename(columns={
    "t1_t1_ra":"ra_des",
    "t1_t1_dec":"dec_des",
    "t1_t1_mag_aper_4_g":"g_mag",
    "t1_t1_mag_aper_4_r":"r_mag",
    "t1_t1_mag_aper_4_i":"i_mag",
    "t1_t1_mag_aper_4_z":"z_mag",
    "t1_t1_mag_aper_4_y":"Y_mag",
    "t1_t1_magerr_aper_4_g":"g_magerr",
    "t1_t1_magerr_aper_4_r":"r_magerr",
    "t1_t1_magerr_aper_4_i":"i_magerr",
    "t1_t1_magerr_aper_4_z":"z_magerr",
    "t1_t1_magerr_aper_4_y":"Y_magerr",
    "japermag4":"J_mag",
    "japermag4err":"J_magerr",
    "ksapermag4":"K_mag",
    "ksapermag4err":"K_magerr",
    "t1_mag_w1_vg":"W1_mag_vg",
    "t1_mag_w2_vg":"W2_mag_vg",
    "t1_fluxlbs_w1":"W1_flux_nMgy",
    "t1_fluxlbs_w2":"W2_flux_nMgy",
    "t1_dfluxlbs_w1":"W1_fluxerr_nMgy",
    "t1_dfluxlbs_w2":"W2_fluxerr_nMgy",
    "ra2000":"ra_vhs",
    "dec2000":"dec_vhs",
    "dist_arcsec":"dist_des_vhs",
    "t1_ra":"ra_wise",
    "t1_dec":"dec_wise",
    "t1_dist_arcsec":"dist_des_wise"
})
neat_df.index.names=['coadd_object_id']
neat_df

ra_des    dec_des      g_mag      r_mag      i_mag  \
coadd_object_id                                                           
914697722        319.119200 -45.090354  18.125549  16.507605  17.101320   
1246635859        32.710920 -12.110579  17.643240  16.183584  16.517576   
984527913        344.403976 -40.240537  15.945573  16.843105  16.362082   
921966171        321.279299 -55.389702  17.825924  16.296925  16.691380   
870486068        299.876297 -53.999905  15.880921  16.138315  16.533918   
...                     ...        ...        ...        ...        ...   
890672347        307.551962 -49.679314  28.945040  24.445572  22.015619   
1117460595        12.880701 -45.886776  26.059227  24.326349  22.029612   
1311137024        25.655443  -0.712265  24.325096  23.013109  22.148762   
1437291121        60.389445 -14.630045  26.888927  24.858255  22.059185   
1580230294        86.758839 -46.337940  24.081594  22.841181  22.092491   

                     z_mag      Y_mag  g_magerr  r_magerr  i_magerr  z_magerr  \
coadd_object_id                                                                 
914697722        15.075032  14.713349  0.000741  0.000339  0.000560  0.000400   
1246635859       15.214206  14.503883  0.000647  0.000289  0.000469  0.000453   
984527913        15.239571  15.196484  0.000301  0.000591  0.000548  0.000297   
921966171        15.280548  14.936352  0.000651  0.000296  0.000500  0.000335   
870486068        15.384865  15.597589  0.000244  0.000255  0.000429  0.000242   
...                    ...        ...       ...       ...       ...       ...   
890672347        20.999533  20.731770  5.077159  0.105080  0.019959  0.017075   
1117460595       20.999763  20.902056  0.355723  0.092730  0.020485  0.016868   
1311137024       20.999807  21.066023  0.072679  0.027314  0.023289  0.014506   
1437291121       20.999891  20.687372  0.625257  0.124696  0.016738  0.012384   
1580230294       20.999891  21.399715  0.052119  0.022123  0.020236  0.014069   

                 Y_magerr   dec_wise  W1_fluxerr_nMgy  W2_fluxerr_nMgy  \
coadd_object_id                                                          
914697722        0.000381 -45.090322        26.233025        62.808285   
1246635859       0.000547 -12.110539        24.390047        59.220913   
984527913        0.000755 -40.240554        13.105026        35.000120   
921966171        0.000474 -55.389683        19.049904        46.321644   
870486068        0.001358 -53.999912        12.821824        34.966680   
...                   ...        ...              ...              ...   
890672347        0.047391 -49.679354         7.069963         0.000000   
1117460595       0.040521 -45.886752         6.069972         0.000000   
1311137024       0.052359  -0.712180         7.968876        32.033960   
1437291121       0.035437 -14.629854         7.172823         0.000000   
1580230294       0.063653 -46.337891         5.628012        20.959763   

                 W1_flux_nMgy  W2_flux_nMgy  W1_mag_vg  W2_mag_vg     ra_wise  \
coadd_object_id                                                                 
914697722         23075.97900   26846.29100  11.592440  11.428191  319.119198   
1246635859        19631.18200   23035.37100  11.768310  11.594521   32.710872   
984527913          4355.65600    4089.07280  13.403144  13.471869  344.403963   
921966171         11148.60700   11932.68000  12.382002  12.308417  321.279281   
870486068          3824.09700    3655.99150  13.542056  13.595168  299.876326   
...                       ...           ...        ...        ...         ...   
890672347            67.90782       0.00000  17.809786        inf  307.551566   
1117460595           66.67445       0.00000  17.847363        inf   12.880809   
1311137024          276.67416     365.60960  16.391867  16.088333   25.655491   
1437291121           51.82774       0.00000  18.162922        inf   60.389366   
1580230294          331.80670     865.44507  16.192630  

### Converting Magnitudes to Flux
Useful for certain SED fitting programs

In [45]:
def mag_to_flx(mag):
    """
    Input: an AB magnitude
    Output: a flux in microjansky (muJy)
    Works with np arrays and pd Series
    """
    return 10**(29-(48.60/2.5)) * 10**(-mag/2.5)

print(mag_to_flx(0) * 1e-6) # 3631 Jy

3630.7805477010024


### Converting Vega to AB

The WISE magnitudes are in the Vega magnitude system; the AB system corresponds better to $F_\nu$. $m_{AB} = m_{Vega} + \Delta m$ where $\Delta m$ [depends on the band](https://wise2.ipac.caltech.edu/docs/release/allsky/expsup/sec4_4h.html):

|Band|$\Delta m$|
|---|---|
|W1|2.699|
|W2|3.339|

### Errors in Magnitude and Flux
The errors use the following relation:
$$
m = -2.5\log(f/f_0)
\qquad\Rightarrow\qquad
|\text{d}m| = 2.5 \log(e) \frac{\text{d}f}{f} \approx 1.09 \frac{\text{d}f}{f}
$$
For DES and VHS, $\text{d}m$ is given; for WISE, $\text{d}f$ and $f$ are given (both in nanomaggies, which seem to be simply proportional to Jy, see [here](https://www.sdss3.org/dr8/algorithms/magnitudes.php)).

In [46]:
wise_delta_m = {'W1':2.699, 'W2':3.339}
for band in ['W1','W2']:
    neat_df[f'{band}_mag'] = neat_df[f'{band}_mag_vg'] + wise_delta_m[band]
    neat_df[f'{band}_magerr'] = 2.5*np.log10(np.e) * neat_df[f'{band}_fluxerr_nMgy'] / neat_df[f'{band}_flux_nMgy']

all_bands = ['g','r','i','z','Y','J','K', 'W1','W2']
for band in all_bands:
    flx = mag_to_flx(neat_df[f'{band}_mag'])
    neat_df[f'{band}_flux'] = flx
    neat_df[f'{band}_fluxerr'] = flx*neat_df[f'{band}_magerr'] / (2.5*np.log10(np.e))

neat_df

ra_des    dec_des      g_mag      r_mag      i_mag  \
coadd_object_id                                                           
914697722        319.119200 -45.090354  18.125549  16.507605  17.101320   
1246635859        32.710920 -12.110579  17.643240  16.183584  16.517576   
984527913        344.403976 -40.240537  15.945573  16.843105  16.362082   
921966171        321.279299 -55.389702  17.825924  16.296925  16.691380   
870486068        299.876297 -53.999905  15.880921  16.138315  16.533918   
...                     ...        ...        ...        ...        ...   
890672347        307.551962 -49.679314  28.945040  24.445572  22.015619   
1117460595        12.880701 -45.886776  26.059227  24.326349  22.029612   
1311137024        25.655443  -0.712265  24.325096  23.013109  22.148762   
1437291121        60.389445 -14.630045  26.888927  24.858255  22.059185   
1580230294        86.758839 -46.337940  24.081594  22.841181  22.092491   

                     z_mag      Y_mag  g_magerr  r_magerr  i_magerr  z_magerr  \
coadd_object_id                                                                 
914697722        15.075032  14.713349  0.000741  0.000339  0.000560  0.000400   
1246635859       15.214206  14.503883  0.000647  0.000289  0.000469  0.000453   
984527913        15.239571  15.196484  0.000301  0.000591  0.000548  0.000297   
921966171        15.280548  14.936352  0.000651  0.000296  0.000500  0.000335   
870486068        15.384865  15.597589  0.000244  0.000255  0.000429  0.000242   
...                    ...        ...       ...       ...       ...       ...   
890672347        20.999533  20.731770  5.077159  0.105080  0.019959  0.017075   
1117460595       20.999763  20.902056  0.355723  0.092730  0.020485  0.016868   
1311137024       20.999807  21.066023  0.072679  0.027314  0.023289  0.014506   
1437291121       20.999891  20.687372  0.625257  0.124696  0.016738  0.012384   
1580230294       20.999891  21.399715  0.052119  0.022123  0.020236  0.014069   

                 Y_magerr   dec_wise  W1_fluxerr_nMgy  W2_fluxerr_nMgy  \
coadd_object_id                                                          
914697722        0.000381 -45.090322        26.233025        62.808285   
1246635859       0.000547 -12.110539        24.390047        59.220913   
984527913        0.000755 -40.240554        13.105026        35.000120   
921966171        0.000474 -55.389683        19.049904        46.321644   
870486068        0.001358 -53.999912        12.821824        34.966680   
...                   ...        ...              ...              ...   
890672347        0.047391 -49.679354         7.069963         0.000000   
1117460595       0.040521 -45.886752         6.069972         0.000000   
1311137024       0.052359  -0.712180         7.968876        32.033960   
1437291121       0.035437 -14.629854         7.172823         0.000000   
1580230294       0.063653 -46.337891         5.628012        20.959763   

                 W1_flux_nMgy  W2_flux_nMgy  W1_mag_vg  W2_mag_vg     ra_wise  \
coadd_object_id                                                                 
914697722         23075.97900   26846.29100  11.592440  11.428191  319.119198   
1246635859        19631.18200   23035.37100  11.768310  11.594521   32.710872   
984527913          4355.65600    4089.07280  13.403144  13.471869  344.403963   
921966171         11148.60700   11932.68000  12.382002  12.308417  321.279281   
870486068          3824.09700    3655.99150  13.542056  13.595168  299.876326   
...                       ...           ...        ...        ...         ...   
890672347            67.90782       0.00000  17.809786        inf  307.551566   
1117460595           66.67445       0.00000  17.847363        inf   12.880809   
1311137024          276.67416     365.60960  16.391867  16.088333   25.655491   
1437291121           51.82774       0.00000  18.162922        inf   60.389366   
1580230294          331.80670     865.44507  16.192630  

### Confusion Sensitivity (at $3\sigma$)
If an object is detected *below* a $3\sigma$ level, the magnitude and error are set to `np.inf`, the flux is set to 0, and the flux error is set to the 3$\sigma$ sensitivity. The sensitivities of the three data sources (DES, VHS, and WISE) are provided in different ways:

#### DES

The 10$\sigma$ AB magnitude depths are given [here](https://iopscience.iop.org/article/10.3847/1538-4365/ac00b3/pdf) in Table 3 Row 3:

|Band|$10\sigma$ Sensitivity / mag$_{AB}$|
|---|---|
|g|24.7|
|r|24.4|
|i|23.8|
|z|23.1|
|Y|21.7|

The relationship between magnitudes at different significance levels is
\begin{align*}
m_{n_1\sigma} - m_{n_2\sigma} 
&= -2.5\log(F_{n_1\sigma} / F_{n_2\sigma})\\
&\equiv -2.5 \log(n_1/n_2)\\
\Rightarrow m_{n_1\sigma}
&= m_{n_2\sigma} - 2.5 \log(n_1/n_2)
\end{align*}

So $m_{3\sigma} = m_{10\sigma} - 2.5 \log(3/10)$.

#### VHS

The (median) $5\sigma$ depths are given [here](http://www.eso.org/rm/api/v1/public/releaseDescriptions/144), just above Table 4, as J:20.8, K:20.0. These are $5\sigma$, so $m_{3\sigma} = m_{5\sigma} - 2.5\log(3/5)$

#### WISE

The $m_{n\sigma}$ are not given for each band, but the $5\sigma$ *flux* sensitivities are:

W1: $F_{\nu, 5\sigma} = 0.08$ mJy

W2: $F_{\nu, 5\sigma} = 0.11$ mJy

In [48]:
des_m_3s = np.array([24.7,24.4,23.8,23.1,21.7]) - 2.5*np.log10(3/10)
des_F_3s = mag_to_flx(des_m_3s)
vhs_m_3s = np.array([20.8,20.0]) - 2.5*np.log10(3/5)
vhs_F_3s = mag_to_flx(vhs_m_3s)
wise_F_3s = np.array([0.08e-3, 0.11e-3])
F_3s = {
    'g':des_F_3s[0],
    'r':des_F_3s[1],
    'i':des_F_3s[2],
    'z':des_F_3s[3],
    'Y':des_F_3s[4],
    'J':vhs_F_3s[0],
    'K':vhs_F_3s[1],
    'W1':wise_F_3s[0],
    'W2':wise_F_3s[1]
}

for idd in tqdm(neat_df.index):
    for band in all_bands:
        sensitivity = F_3s[band]
        if neat_df.at[idd, f'{band}_flux'] < sensitivity:
            neat_df.at[idd, f'{band}_mag'] = 99.
            neat_df.at[idd, f'{band}_magerr'] = 99.
            neat_df.at[idd, f'{band}_flux'] = 0.
            neat_df.at[idd, f'{band}_fluxerr'] = sensitivity
neat_df

100%|████████████████████████████████████████████████████████████████████████| 151629/151629 [00:28<00:00, 5316.70it/s]


ra_des    dec_des      g_mag      r_mag      i_mag  \
coadd_object_id                                                           
914697722        319.119200 -45.090354  18.125549  16.507605  17.101320   
1246635859        32.710920 -12.110579  17.643240  16.183584  16.517576   
984527913        344.403976 -40.240537  15.945573  16.843105  16.362082   
921966171        321.279299 -55.389702  17.825924  16.296925  16.691380   
870486068        299.876297 -53.999905  15.880921  16.138315  16.533918   
...                     ...        ...        ...        ...        ...   
890672347        307.551962 -49.679314  99.000000  24.445572  22.015619   
1117460595        12.880701 -45.886776  99.000000  24.326349  22.029612   
1311137024        25.655443  -0.712265  24.325096  23.013109  22.148762   
1437291121        60.389445 -14.630045  99.000000  24.858255  22.059185   
1580230294        86.758839 -46.337940  24.081594  22.841181  22.092491   

                     z_mag      Y_mag   g_magerr  r_magerr  i_magerr  \
coadd_object_id                                                        
914697722        15.075032  14.713349   0.000741  0.000339  0.000560   
1246635859       15.214206  14.503883   0.000647  0.000289  0.000469   
984527913        15.239571  15.196484   0.000301  0.000591  0.000548   
921966171        15.280548  14.936352   0.000651  0.000296  0.000500   
870486068        15.384865  15.597589   0.000244  0.000255  0.000429   
...                    ...        ...        ...       ...       ...   
890672347        20.999533  20.731770  99.000000  0.105080  0.019959   
1117460595       20.999763  20.902056  99.000000  0.092730  0.020485   
1311137024       20.999807  21.066023   0.072679  0.027314  0.023289   
1437291121       20.999891  20.687372  99.000000  0.124696  0.016738   
1580230294       20.999891  21.399715   0.052119  0.022123  0.020236   

                 z_magerr  Y_magerr   dec_wise  W1_fluxerr_nMgy  \
coadd_object_id                                                   
914697722        0.000400  0.000381 -45.090322        26.233025   
1246635859       0.000453  0.000547 -12.110539        24.390047   
984527913        0.000297  0.000755 -40.240554        13.105026   
921966171        0.000335  0.000474 -55.389683        19.049904   
870486068        0.000242  0.001358 -53.999912        12.821824   
...                   ...       ...        ...              ...   
890672347        0.017075  0.047391 -49.679354         7.069963   
1117460595       0.016868  0.040521 -45.886752         6.069972   
1311137024       0.014506  0.052359  -0.712180         7.968876   
1437291121       0.012384  0.035437 -14.629854         7.172823   
1580230294       0.014069  0.063653 -46.337891         5.628012   

                 W2_fluxerr_nMgy  W1_flux_nMgy  W2_flux_nMgy  W1_mag_vg  \
coadd_object_id                                                           
914697722              62.808285   23075.97900   26846.29100  11.592440   
1246635859             59.220913   19631.18200   23035.37100  11.768310   
984527913              35.000120    4355.65600    4089.07280  13.403144   
921966171              46.321644   11148.60700   11932.68000  12.382002   
870486068              34.966680    3824.09700    3655.99150  13.542056   
...                          ...           ...           ...        ...   
890672347               0.000000      67.90782       0.00000  17.809786   
1117460595              0.000000      66.67445       0.00000  17.847363   
1311137024             32.033960     276.67416     365.60960  16.391867   
1437291121              0.000000      51.82774       0.00000  18.162922   
1580230294             20.959763     331.80670     865.44507  16.192630   

                 W2_mag_vg     ra_wise  dist_des_wise    dec_vhs      J_mag  \
coadd_object_id                                                               
914697722        11.428191  319.119198       0.117144 -45.090308  12.526873   
1246635859       11.594

## Cuts
The above dataframe represents a liberal cut in DES data and an X-match (1") to VHS and WISE detections. We begin with 151,629 items.

We then make the following cuts:

* (W1, W2) $\neq 99$. This ensures that the object was in fact detected in WISE.
* (dW1, dW2) $<2.5\log (e)/3$. This ensures a signal-to-noise ratio of greater than 3.
* W1 - W2 $>0.5$ (Vega mags). This removes many of the contaminants (e.g. dwarf stars).

2,910 objects remain.

In [59]:
cut_df = neat_df[
    (neat_df['W1_mag']!=99.) &
    (neat_df['W2_mag']!=99.) &
    (neat_df['W1_magerr'] < 2.5*np.log10(np.e) / 3) & 
    (neat_df['W2_magerr'] < 2.5*np.log10(np.e) / 3) &
    (neat_df['W1_mag_vg'] - neat_df['W2_mag_vg'] > 0.5)
]
cut_df

ra_des    dec_des      g_mag      r_mag      i_mag  \
coadd_object_id                                                           
931684428        324.850726 -42.589015  16.599474  16.393660  16.711504   
1556041733        81.563282 -41.218403  16.374800  16.465633  17.463329   
979002481        342.171547 -51.164810  15.673114  16.068356  17.093698   
995403105        347.111985 -53.411194  17.393402  16.070789  17.274448   
979097465        342.171892 -45.439281  16.905516  16.111046  17.142202   
...                     ...        ...        ...        ...        ...   
887791189        304.571526 -64.443206  99.000000  25.289423  22.204578   
1697986932        79.667443 -54.549900  99.000000  24.895716  22.037481   
1011682001       352.995897 -51.519823  24.987549  23.203817  22.057419   
959369723        335.415977 -47.871207  23.950171  22.853050  22.039396   
1580230294        86.758839 -46.337940  24.081594  22.841181  22.092491   

                     z_mag      Y_mag   g_magerr  r_magerr  i_magerr  \
coadd_object_id                                                        
931684428        15.676861  15.332111   0.000311  0.000273  0.000417   
1556041733       15.783970  15.415310   0.000273  0.000299  0.000593   
979002481        15.902316  15.743353   0.000322  0.000266  0.000554   
995403105        15.892651  14.998402   0.000491  0.000231  0.000594   
979097465        16.109114  15.093253   0.000371  0.000255  0.000554   
...                    ...        ...        ...       ...       ...   
887791189        20.991449  20.781302  99.000000  0.264862  0.028434   
1697986932       20.994463  20.771387  99.000000  0.151750  0.019640   
1011682001       20.994617  20.924448   0.146228  0.040612  0.024272   
959369723        20.998436  20.973963   0.056980  0.027209  0.023609   
1580230294       20.999891  21.399715   0.052119  0.022123  0.020236   

                 z_magerr  Y_magerr   dec_wise  W1_fluxerr_nMgy  \
coadd_object_id                                                   
931684428        0.000259  0.000570 -42.589000        30.999132   
1556041733       0.000254  0.000541 -41.218417        13.246780   
979002481        0.000335  0.000668 -51.164813        29.458530   
995403105        0.000374  0.000368 -53.411204        21.238377   
979097465        0.000358  0.000575 -45.439294        19.046530   
...                   ...       ...        ...              ...   
887791189        0.017894  0.054016 -64.443417         6.983442   
1697986932       0.014022  0.038908 -54.549814         4.191251   
1011682001       0.018400  0.047916 -51.519773         7.501736   
959369723        0.014383  0.047931 -47.871210         7.654265   
1580230294       0.014069  0.063653 -46.337891         5.628012   

                 W2_fluxerr_nMgy  W1_flux_nMgy  W2_flux_nMgy  W1_mag_vg  \
coadd_object_id                                                           
931684428              95.513985  33422.754000   74348.87500  11.189637   
1556041733             33.451920   3525.532200    5923.62200  13.637177   
979002481              95.613100  30659.105000   77761.86000  11.282861   
995403105              52.965280  10385.307000   18389.44100  12.462234   
979097465              46.550224   6881.079600   12015.87800  12.910409   
...                          ...           ...           ...        ...   
887791189              26.935120     76.591940     153.11038  17.715685   
1697986932             16.157791     53.300095      95.89126  18.161716   
1011682001             26.374840    393.562230     723.80360  16.007532   
959369723              28.458288    314.657070     656.23520  16.243427   
1580230294             20.959763    331.806700     865.44507  16.192630   

                 W2_mag_vg     ra_wise  dist_des_wise    dec_vhs      J_mag  \
coadd_object_id                                                               
931684428        10.321410  324.850690       0.109317 -42.588989  13.917383   
1556041733       13.065

In [11]:
cut_wise_df = wise_df[(wise_df['mag_w1_ab']!=np.inf) &
                      (wise_df['mag_w2_ab']!=np.inf)]
N_SIG=3
mag_sens = mag_sensitivity(n_sigma=N_SIG)
cut_wise_df = cut_wise_df[(cut_wise_df['mag_w1_ab'] < mag_sens[0]) &
                          (cut_wise_df['mag_w2_ab'] < mag_sens[1])]
error = 2.5*np.log(np.e) / N_SIG
cut_wise_df = cut_wise_df[(cut_wise_df['dmag_w1'] < error) &
                          (cut_wise_df['dmag_w2'] < error)]
cut_wise_df = cut_wise_df[cut_wise_df['mag_w1_vg'] - cut_wise_df['mag_w2_vg'] > .5]

cut_wise_df

t1_ra     t1_dec  t1_mag_aper_4_g  t1_mag_aper_4_r  \
t1_coadd_object_id                                                            
931684428           324.850726 -42.589015        16.599474        16.393660   
1556041733           81.563282 -41.218403        16.374800        16.465633   
995403105           347.111985 -53.411194        17.393402        16.070789   
979002481           342.171547 -51.164810        15.673114        16.068356   
1202849140           27.093035 -27.973787        15.960557        16.349682   
...                        ...        ...              ...              ...   
1128468683           15.525550 -34.172503        23.951994        23.005493   
1289118931           38.137501   4.673760        25.327623        23.204370   
1045416151            2.387374  -5.157882        23.461046        22.605087   
1057302608            4.478374 -52.381591        23.591862        22.835726   
1218053653           29.301543 -33.395768        24.663685        23.259235   

                    t1_mag_aper_4_i  t1_mag_aper_4_z  t1_mag_aper_4_y  \
t1_coadd_object_id                                                      
931684428                 16.711504        15.676861        15.332111   
1556041733                17.463329        15.783970        15.415310   
995403105                 17.274448        15.892651        14.998402   
979002481                 17.093698        15.902316        15.743353   
1202849140                17.337967        16.125484        15.436979   
...                             ...              ...              ...   
1128468683                22.130493        20.996181        20.834501   
1289118931                22.021729        20.985817        20.815287   
1045416151                22.004913        20.987995        21.133406   
1057302608                22.084969        20.990633        20.833904   
1218053653                22.124104        20.991196        20.782921   

                    t1_magerr_aper_4_g  t1_magerr_aper_4_r  \
t1_coadd_object_id                                           
931684428                     0.000311            0.000273   
1556041733                    0.000273            0.000299   
995403105                     0.000491            0.000231   
979002481                     0.000322            0.000266   
1202849140                    0.000245            0.000256   
...                                ...                 ...   
1128468683                    0.053829            0.030031   
1289118931                    0.311004            0.050187   
1045416151                    0.032261            0.020582   
1057302608                    0.038924            0.024828   
1218053653                    0.124541            0.040713   

                    t1_magerr_aper_4_i  ...   fluxlbs_w1    fluxlbs_w2  \
t1_coadd_object_id                      ...                              
931684428                     0.000417  ...  33422.75400   74348.87500   
1556041733                    0.000593  ...   3525.53220    5923.62200   
995403105                     0.000594  ...  10385.30700   18389.44100   
979002481                     0.000554  ...  30659.10500   77761.86000   
1202849140                    0.000535  ...  47139.78500  112593.85000   
...                                ...  ...          ...           ...   
1128468683                    0.023195  ...    223.74790     436.23334   
1289118931                    0.024924  ...    558.80350     963.53200   
1045416151                    0.018922  ...    310.26007     781.63760   
1057302608                    0.021894  ...    305.04416     754.20510   
1218053653                    0.023882  ...    296.36215     498.33680   

                    dfluxlbs_w1  dfluxlbs_w2  mag_w1_vg  mag_w2_vg  mag_w1_ab  \
t1_coadd_object_id                                                              
931684428             30.999132    95.513985  11.189637  10.321410  13.888637   
1556041733            13.246780   

In [12]:
cut_wise_df.to_csv('objs_2910.csv')

## Saving Data

However, not all of the above 2,910 objects will have images in DES - many will have failed in the pipeline for various reasons (e.g. images not available in all bands, images on the edge of DES tiles, etc.). We now find the objects and images which are not only present in WISE, but also have complete DES imaging, as these will be the ones the NN works with.

This removes a further 173 objects, giving a final 2737 images to work with.

In [14]:
all_1e6_fl = np.load('/data/beegfs/astro-storage/groups/walter/byrne/pipe/all_1e6.npz')
all_imgs = all_1e6_fl['imgs']
all_ids = all_1e6_fl['ids']

In [15]:
present_in_wise = np.in1d(all_ids, cut_wise_df.index.to_numpy())
wise_imgs = all_imgs[present_in_wise]
wise_ids = all_ids[present_in_wise]
len(wise_ids)

2737

In [14]:
np.savez_compressed('/data/beegfs/astro-storage/groups/walter/byrne/pipe/wise_imgs',
                   imgs = wise_imgs,
                   ids = wise_ids)

In [21]:
cut_wise_df.loc[wise_ids].to_csv('/data/beegfs/astro-storage/groups/walter/byrne/orig_plus_wise_cuts.csv')